src/Microsoft.AspNetCore.Mvc.Core/Internal/FileContentResultExecutor.cs

```diff
@@ -15,15 +17,54 @@ public FileContentResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileContentResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            RangeItemHeaderValue range;
+            long rangeLength;
+            if (result.LastModified.HasValue)
```

Comment:
e-tags and last-modified can be processed independently, you should check if either of them are set.